### ISSUE TITLE
Residual Target in MLMG solver 

### ISSUE THREAD

**Issue by ankithadas** [AMReX-Codes/amrex#3590]
Hey everyone,

I'd like to discuss a potential breaking change in MLMG residual target calculation. Currently, MLMG offers the option to 'always use b norm' or select the greater value between RHS norm and Residual norm. Consequently, there's no mechanism to consistently use the standard Residual norm (|b - Ax|) for residual target calculation. This could be crucial for specific calculations. The proposed breaking change involves removing ```void setAlwaysUseBNorm(int flag)``` and replacing it with something like ```void setNormType(NormType n = NormType::Default)```, where ```NormType``` is an enum."

Any thoughts ?

Kind regards,
Ankith

**Comment by WeiqunZhang** [comment]
Note that the b norm can also be regarded as the norm of `b - A * x_initial_guess`, where the initial guess is zero.

 The issue of using `b - A * x_initial_guess` alone is if the initial guess is too good, then the target will be unachievable.

**Comment by ankithadas** [comment]
True, but if absolute tolerance is set to a non-zero value, a good enough guess would exit out from MLMG. My proposed change is to either 
1. Use the current default behaviour where max_norm is calculated using the greater of residual norm (|b - A x_initial) and b norm (|b|)
2. Use always b norm 
3. Use always residual norm 

The behaviour can be set by the user.

**Comment by asalmgren** [comment]
Closing this issue due to lack of activity.  Please re-open it if you feel there is is still an issue!
